packages/11ty/_plugins/iiif/process/createImage.js

```diff
@@ -15,32 +15,43 @@ module.exports = (eleventyConfig) => {
   /**
    * Creates an image in the output directory with the name `${name}${ext}`
    *
-   * @param  {String} input Input file path
+   * @param  {Object} figure Figure entry data from `figures.yaml`
    * @param  {Object} transformation A transformation item from `iiif/config.js#imageTransformations`
    * @property  {String} name The name of the file
    * @property  {Object} resize Resize options for `sharp`
    */
-  return async (filename, transformation = {}, options) => {
+  return async (figure, transformation = {}, options) => {
     const { debug, lazy } = options
 
-    const { ext, name } = path.parse(filename)
+    const { region, src } = figure
+    const { ext, name } = path.parse(src)
+    const { resize } = transformation
     const format = formats.find(({ input }) => input.includes(ext))
-    const inputPath = path.join(inputDir, filename)
+    const inputPath = path.join(inputDir, src)
     const outputPath = path.join(outputRoot, outputDir, name, `${transformation.name}${format.output}`)
 
     fs.ensureDirSync(path.parse(outputPath).dir)
 
+    const service = sharp(inputPath)
+    service.crop = function (region) {
+      if (!region) return this
+      const [ top, left, width, height ] = region.split(',').map((item) => parseFloat(item.trim()))
+      service.extract({ top, left, width, height })
+      return this
+    }
+
     if (!lazy || !fs.pathExistsSync(outputPath)) {
       if (debug) {
-        info(`Created ${filename}`)
+        info(`Created ${src}`)
       }
       try {
-        return await sharp(inputPath)
-          .resize(transformation.resize)
+        return await service
+          .crop(region)
+          .resize(resize)
```

Comment:
@mphstudios I could use some feedback on this pattern - not sure this is exactly right